lane.py

```diff
@@ -1707,12 +1730,28 @@ def _featured_works_with_lanes(
             return
 
         # Ask the search engine for works from every lane we're given.
+
+        # NOTE: At the moment, every WorkList in the system can be
+        # generated using an Elasticsearch query. That is, there are
+        # no subclasses of the DatabaseExclusiveWorkList class defined
+        # in circulation/api/lanes.py. If that ever changes, we'll
+        # need to change this code.
+        #
+        # The simplest change would probably be to return a dictionary
+        # mapping WorkList to Works and let the caller figure out the
+        # ordering. In fact, we could start doing that now.
+        queries = []
         for lane in lanes:
-            adapted = lane.overview_facets(_db, facets)
-            for work in lane.works(
-                _db, adapted, pagination, search_engine=search_engine,
-                debug=debug
-            ):
+            overview_facets = lane.overview_facets(_db, facets)
+            from external_search import Filter
+            filter = Filter.from_worklist(_db, lane, overview_facets)
+            queries.append((None, filter, pagination))
+        resultsets = list(search_engine.query_works_multi(queries))
+        works = self.works_for_resultsets(_db, resultsets)
+
+        for i, lane in enumerate(lanes):
+            results = works[i]
+            for work in results:
```

Comment:
So this will return a work and its lane for every work that was found?

Reply:
Yes, that's right. I don't think we need the 2-tuples anymore -- we could just yield `(lane, results)` -- but I'd rather make that change in a separate branch.